routes/verify.go

```diff
@@ -251,6 +252,208 @@ func (fes *APIServer) validatePhoneNumberNotAlreadyInUse(phoneNumber string, use
+func (fes *APIServer) getCaptchaRewardNanosFromGlobalState() (uint64, error) {
+	dbKey := GlobalStateKeyForCaptchaRewardAmountNanos()
+
+	rewardNanosBytes, err := fes.GlobalState.Get(dbKey)
+	if err != nil {
+		return 0, fmt.Errorf(
+			"getCaptchaRewardNanosFromGlobalState: Problem with Get: %v", err)
+	}
+
+	rewardNanos, err := lib.ReadUvarint(bytes.NewReader(rewardNanosBytes))
+
+	return rewardNanos, nil
+}
+
+// putCaptchaRewardNanosInGlobalState puts the amount of nanos to reward for a successful captcha in global state.
+func (fes *APIServer) putCaptchaRewardNanosInGlobalState(rewardNanos uint64) error {
+	dbKey := GlobalStateKeyForCaptchaRewardAmountNanos()
+
+	rewardNanosBytes := lib.UintToBuf(rewardNanos)
+
+	if err := fes.GlobalState.Put(dbKey, rewardNanosBytes); err != nil {
+		return fmt.Errorf(
+			"putCaptchaRewardNanosInGlobalState: Problem with Put: %v", err)
+	}
+
+	return nil
+}
+
+// verifyHCaptchaTokenAndSendStarterDESO verifies the captcha token and sends the starter DESO to the user.
+func (fes *APIServer) verifyHCaptchaTokenAndSendStarterDESO(token string, publicKeyBase58Check string) (txnHashHex string, err error) {
+	// Verify the token with hCaptcha.
+	verificationSuccess, err := fes.verifyHCaptchaToken(token)
+
+	if err != nil {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: Error verifying captcha: %v", err)
+	}
+
+	if !verificationSuccess {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: Captcha verification failed")
+	}
+
+	// Ensure the user has not already received the starter DESO for submitting a successful captcha.
+	userMetadata, err := fes.getUserMetadataFromGlobalState(publicKeyBase58Check)
+	if err != nil {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: Problem with getUserMetadataFromGlobalState: %v", err)
+	}
+
+	if userMetadata.LastHcaptchaBlockHeight != 0 {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: LastHcaptchaBlockHeight is already set")
+	}
+
+	if fes.Config.StarterDESOSeed == "" {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: Starter DESO seed not set")
+	}
+
+	// Retrieve the amount of nanos to reward for a successful captcha.
+	amountToSendNanos, err := fes.getCaptchaRewardNanosFromGlobalState()
+	if err != nil {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: Problem with getCaptchaRewardNanosFromGlobalState: %v", err)
+	}
+
+	// Update the user's metadata to indicate that they have received the starter DESO.
+	lastBlockheight := fes.blockchain.BlockTip().Height
+	userMetadata.LastHcaptchaBlockHeight = lastBlockheight
+	userMetadata.HcaptchaShouldCompProfileCreation = true
+
+	if err = fes.putUserMetadataInGlobalState(userMetadata); err != nil {
+		return "", fmt.Errorf("HandleCaptchaVerificationRequest: Problem with putUserMetadataInGlobalState: %v", err)
+	}
+
+	// Send the starter DESO to the user.
+	var txnHash *lib.BlockHash
+	publicKeyBytes, _, err := lib.Base58CheckDecode(publicKeyBase58Check)
```

Comment:
Handle this error? also, maybe do this earlier in the function (or parse it in the calling function)? We probably want to have all inputs verified before we call this function.